sdk/resource/resource.go

```diff
@@ -17,30 +17,57 @@
 package resource
 
 import (
-	"reflect"
+	"sort"
+	"strings"
 
 	"go.opentelemetry.io/otel/api/core"
 )
 
 // Resource describes an entity about which identifying information and metadata is exposed.
 type Resource struct {
-	labels map[core.Key]core.Value
+	str    string
+	sorted []core.KeyValue
+	keys   map[core.Key]struct{}
```

Comment:
I'm slightly puzzled - why do we need to abandon typing and go to storing struct{}?

Reply:
This is only used in the `Merge` func where we need to know if one Resource contains a key that the other already has. It's a smaller memory footprint without the `Value`s (but not a huge one), and I was thinking using the unofficial set representation Go would indicate the purpose of the field.

I'm happy keeping the original if there are plans to use the associated values.

Reply:
suggest using `keySet` or similar then to convey that meaning explicitly.

Reply:
sounds good, will do.